### PR TITLE
Parse require-style imports for JS & TS

### PR DESCRIPTION
**Pull request by nicoespeon**
As discussed, I finally adapted the parsers to handle the `require()` imports.

I added tests to show it's working fine. I also added TS tests that were missing (a duplicate from JS ones at this point).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -14,12 +14,12 @@ and get it running as a standalone tool (see [README](README.md) for a step by s
...
 ## The following code style/convention/tooling details currently exist for this project.
 
+- The code style in this project is based on the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guide for python code, with a minimal customization of those rules for pylint.
+
 - This project uses [pylint](https://pypi.org/project/pylint/) with the following rule modification
 ```
 pylint --msg-template "{path}:{line}:{column}:{category}:{symbol} - {msg}" --disable=all --enable=F,E,unneeded-not,invalid-name,unidiomatic-typecheck,too-many-lines,multiple-imports,comparison-with-itself,cyclic-import,too-many-ancestors,too-many-branches,too-many-statements,too-many-nested-blocks,consider-using-join,bad-classmethod-argument,unused-argument,protected-access,abstract-method,unreachable,duplicate-key,unnecessary-semicolon,global-variable-not-assigned,unused-variable,binary-op-exception,bad-format-string,anomalous-backslash-in-string,bad-open-mode emerge 
 ```
 
-- The code style in this project is based on the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guide for python code, with a minimal customization of those rules for pylint (details in -> Contributing.md)
```
Comment (nicoespeon) [inline CONTRIBUTING.md:22]: I noticed this was referring to the previous point, so I updated it.

File: emerge/languages/javascriptparser.py [hunk 2]
```diff
@@ -113,63 +114,71 @@ def _add_imports_to_result(self, result: AbstractResult, analysis):
         filtered_list_no_comments = self.preprocess_file_content_and_generate_token_list_by_mapping(source_string_no_comments, self._token_mappings)
 
         for _, obj, following in self._gen_word_read_ahead(filtered_list_no_comments):
-            if obj == JavaScriptParsingKeyword.IMPORT.value:
-                read_ahead_string = self.create_read_ahead_string(obj, following)
+            if obj != JavaScriptParsingKeyword.IMPORT.value and obj != JavaScriptParsingKeyword.REQUIRE.value:
```
Comment (nicoespeon) [inline emerge/languages/javascriptparser.py:117]: At first, I wanted to extract a method, but I realized I was inside a loop. I made it work by early-returning for cases that won't match.

Then, the difference is the `expression_to_match` that would be different regarding the import keyword used.

Comment (glato) [reply emerge/languages/javascriptparser.py:117]: Nice way of dealing with both import cases 👍

**Review by glato — COMMENTED** [review]
@nicoespeon Great work, thanks for the big contribution!
